sdks/java/fn-execution/src/test/java/org/apache/beam/sdk/fn/data/BeamFnDataGrpcMultiplexerTest.java

```diff
@@ -61,37 +63,64 @@
 
   @Test
   public void testOutboundObserver() {
-    Collection<BeamFnApi.Elements> values = new ArrayList<>();
-    BeamFnDataGrpcMultiplexer multiplexer = new BeamFnDataGrpcMultiplexer(
-        DESCRIPTOR,
-        (StreamObserver<BeamFnApi.Elements> inboundObserver)
-            -> TestStreams.withOnNext(values::add).build());
+    final Collection<BeamFnApi.Elements> values = new ArrayList<>();
+    BeamFnDataGrpcMultiplexer multiplexer =
+        new BeamFnDataGrpcMultiplexer(
+            DESCRIPTOR,
+            new StreamObserverClientFactory<Elements, Elements>() {
+              @Override
+              public StreamObserver<Elements> outboundObserverFor(
+                  StreamObserver<Elements> inboundObserver) {
+                return TestStreams.withOnNext(
+                        new Consumer<Elements>() {
+                          @Override
+                          public void accept(Elements item) {
+                            values.add(item);
+                          }
+                        })
+                    .build();
+              }
+            });
     multiplexer.getOutboundObserver().onNext(ELEMENTS);
     assertThat(values, contains(ELEMENTS));
   }
 
   @Test
   public void testInboundObserverBlocksTillConsumerConnects() throws Exception {
-    Collection<BeamFnApi.Elements> outboundValues = new ArrayList<>();
-    Collection<BeamFnApi.Elements.Data> inboundValues = new ArrayList<>();
-    BeamFnDataGrpcMultiplexer multiplexer = new BeamFnDataGrpcMultiplexer(
-        DESCRIPTOR,
-        (StreamObserver<BeamFnApi.Elements> inboundObserver)
-            -> TestStreams.withOnNext(outboundValues::add).build());
+    final Collection<BeamFnApi.Elements> outboundValues = new ArrayList<>();
+    final Collection<BeamFnApi.Elements.Data> inboundValues = new ArrayList<>();
+    final BeamFnDataGrpcMultiplexer multiplexer =
+        new BeamFnDataGrpcMultiplexer(
+            DESCRIPTOR,
+            new StreamObserverClientFactory<Elements, Elements>() {
+              @Override
+              public StreamObserver<Elements> outboundObserverFor(
+                  StreamObserver<Elements> inboundObserver) {
+                return TestStreams.withOnNext(
+                        new Consumer<Elements>() {
+                          @Override
+                          public void accept(Elements item) {
+                            outboundValues.add(item);
```

Comment:
`outboundValues::add` instead of `Consumer<Elements>`

Ditto on you choice of lambda vs StreamObserverClientFactory as seen in the existing code.

Reply:
Done.